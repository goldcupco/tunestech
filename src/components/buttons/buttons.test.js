import { render } from '@testing-library/react';

import { withTheme } from '../../__mocks__/withTheme';
import { Button } from './index';

describe('components | buttons', () => {
  test('Render the Button default', () => {
    const { container } = render(withTheme(<Button>Some text stuff</Button>));
    expect(container).toMatchSnapshot();
  });

  test('Render the Button with variant primary', () => {
    const { container } = render(withTheme(<Button variant="primary">Some text stuff</Button>));
    expect(container).toMatchSnapshot();
  });

  test('Render the cursor ponter when prop onClick is set', () => {
    const { container } = render(withTheme(<Button onClick={jest.fn}>Some text stuff</Button>));
    expect(container).toMatchSnapshot();
  });
});
