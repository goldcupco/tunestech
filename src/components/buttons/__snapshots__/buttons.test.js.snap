// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | buttons Render the Button default 1`] = `
.c0 {
  background-color: transparent;
  padding: 10px;
  border-radius: 20px;
  text-transform: uppercase;
  font-weight: 600;
  font-size: 14px;
  border: 1px solid #bdbcbe;
  color: #5fbe1a;
}

<div>
  <button
    class="c0"
  >
    Some text stuff
  </button>
</div>
`;

exports[`components | buttons Render the Button with variant primary 1`] = `
.c0 {
  background-color: transparent;
  padding: 10px;
  border-radius: 20px;
  text-transform: uppercase;
  font-weight: 600;
  font-size: 14px;
  border: 1px solid #5fbe1a;
  background-color: #5fbe1a;
  color: #ffffff;
}

<div>
  <button
    class="c0"
  >
    Some text stuff
  </button>
</div>
`;

exports[`components | buttons Render the cursor ponter when prop onClick is set 1`] = `
.c0 {
  background-color: transparent;
  padding: 10px;
  border-radius: 20px;
  text-transform: uppercase;
  font-weight: 600;
  font-size: 14px;
  cursor: pointer;
  border: 1px solid #bdbcbe;
  color: #5fbe1a;
}

<div>
  <button
    class="c0"
  >
    Some text stuff
  </button>
</div>
`;
